tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Control/AsyncType.fs

```diff
@@ -538,3 +538,23 @@ type AsyncType() =
         with Failure "finish" -> 
             hasThrown <- true
         Assert.True hasThrown
+
+    [<Fact>]
+    member _.WhileBang () =
+        let mutable count = 0
+
+        let asyncCondition = async {
+            return count < 10
+        }
+
+        async {
+            count <- 1
+            
+            while! asyncCondition do
```

Comment:
This doesn't build in the IDE. Is there a way to #if this so that it is not visible when the SDK compiler is used?

Reply:
It is a test case, right?
Add it as a file (in fsproj) without using the <Compile> element.
See e.g. the .fsproj of ComponentTests project on how test case files are added generically - they are linked to be visible in Solution Explorer, but excluded from compilation. And are then only compiled by latest code when the tests actually runs.

Reply:
I would probably add it as a test in the componenttests suite as well.
(and having the code either as standalone .fs file like mentioned above, or inlined as code in a string.

Reply:
I want to run that code, not just compile it. I don't see those kinds of tests in ComponentTests.

Reply:
C:\code\myFsharpFork\tests\FSharp.Compiler.ComponentTests\CompilerOptions\fsc\reflectionfree.fs

This should help you. Please let me know if I can help further, this is definitely a good addition! :)

<img width="403" alt="image" src="https://user-images.githubusercontent.com/46543583/201101415-a765a77f-058d-4c7b-a0bb-75bb62790883.png">